cmd/frontend/graphqlbackend/highlight.go

```
@@ -35,22 +35,46 @@ type HighlightArgs struct {
 	DisableTimeout     bool
 	IsLightTheme       *bool
 	HighlightLongLines bool
-	TreeSitterEnabled  bool
 }
 
 type highlightedFileResolver struct {
-	aborted bool
-	html    template.HTML
-
-	// JSON encoded form of lsiftyped.Document
-	lsif string
+	aborted  bool
+	response *highlight.HighlightedCode
 }
 
 func (h *highlightedFileResolver) Aborted() bool { return h.aborted }
-func (h *highlightedFileResolver) HTML() string  { return string(h.html) }
-func (h *highlightedFileResolver) LSIF() string  { return h.lsif }
+func (h *highlightedFileResolver) HTML() string {
+	html, err := h.response.HTML()
+	if err != nil {
+		return ""
+	}
+
+	return string(html)
+}
+func (h *highlightedFileResolver) LSIF() string {
+	if h.response == nil {
+		return ""
+	}
+
+	marshaller := &jsonpb.Marshaler{
+		EnumsAsInts:  true,
+		EmitDefaults: false,
+	}
+
+	// TODO(tjdevries): We could probably serialize the error, but it wouldn't do anything for now.
+	lsif, err := marshaller.MarshalToString(h.response.LSIF())
+	if err != nil {
+		return "{}"
```

I'd expect this to be an actual error condition in the resolver if we can't do this. Is this expected on this path because we're not sure if we can actually serialize this, or if the return value of `LSIF()` above may be the empty string? On the latter case we should check explicitly and return `{}` prior to failure. On the former case we should throw a hard error as it's not something we're expecting.